### PR TITLE
Add documentation for downward API hugepages

### PR DESCRIPTION
**Pull request by derekwaynecarr**
Document support for hugepages in 1.20 per https://github.com/kubernetes/enhancements/issues/2053

### CODE REVIEW

**Comment by derekwaynecarr** [comment]
I was not sure how to capture an example, hoping that sig-docs can assist on the right location.

**Comment by kbhawkey** [comment]
> 
> 
> I was not sure how to capture an example, hoping that sig-docs can assist on the right location.

Hi @derekwaynecarr .
Are you interested in adding a configuration example (see content/en/examples/pods/inject directory) to this task page?
There are other examples used in this task. You can add the example to the `examples` directory and 
reference this in the page with the `codenew` shortcode, such as

```
{{< codenew file="pods/inject/dapi-volume.yaml" >}}
```

File: content/en/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information.md [hunk 1]
```diff
@@ -210,6 +210,8 @@ variables and `downwardAPI` volumes:
   * A Container's CPU request
   * A Container's memory limit
   * A Container's memory request
+  * A Container's hugepages limit, available since v1.20 (if DownwardAPIHugePages feature gate enabled)
+  * A Container's hugepages request, available since v1.20 (if DownwardAPIHugePages feature gate enabled)
```
Comment (sftim) [inline content/en/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information.md:214]:
```suggestion
  * A Container's hugepages limit (providing that the `DownwardAPIHugePages` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled)
  * A Container's hugepages request (providing that the `DownwardAPIHugePages` [feature gate](/docs/reference/command-line-tools-reference/feature-gates/) is enabled)
```

File: content/en/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information.md [hunk 2]
```diff
@@ -210,6 +210,8 @@ variables and `downwardAPI` volumes:
...
   * A Container's ephemeral-storage limit, available since v1.8.0-beta.0
   * A Container's ephemeral-storage request, available since v1.8.0-beta.0
```
Comment (sftim) [inline content/en/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information.md:216]:
```suggestion
  * A Container's ephemeral-storage limit
  * A Container's ephemeral-storage request
```

Comment (sftim) [reply content/en/docs/tasks/inject-data-application/downward-api-volume-expose-pod-information.md:216]: and I would also revise this to drop the other mentions of version support for long-graduated fields such as `metadata.uid`.

**Review by sftim — COMMENTED** [review]
Here's how I'd do it.

**Comment by sftim** [comment]
Hi @derekwaynecarr 

Does this feature also need a change to https://k8s.io/docs/reference/command-line-tools-reference/feature-gates/ ?

**Comment by annajung** [comment]
/hold cancel
k/k pr merged

@derekwaynecarr With k/k done, it would be really helpful for Sig Docs if you could get the docs into "ready for review" state early -- before kubecon if you can. Thank you!

**Comment by annajung** [comment]
Hey @derekwaynecarr a friendly reminder that this PR ready for review deadline is coming up soon! (**Nov. 23rd**)
When you can, please review the suggestions provided above and work on getting this PR in a ready to be reviewed state. Thank you!

**Comment by annajung** [comment]
Hi @derekwaynecarr today is the deadline for docs to be in ready for review state. Please address Tim's comment above and let us know if this is ready to be reviewed.

**Comment by derekwaynecarr** [comment]
pushed updates per feedback, thanks!

File: content/en/docs/reference/command-line-tools-reference/feature-gates.md [hunk 3]
```diff
@@ -91,6 +91,7 @@ different Kubernetes components.
 | `DevicePlugins` | `true` | Beta | 1.10 | |
 | `DisableAcceleratorUsageMetrics` | `false` | Alpha | 1.19 | 1.19 |
 | `DisableAcceleratorUsageMetrics` | `true` | Beta | 1.20 | 1.22 |
+| `DownwardAPIHugePages` | `false` | Alpha | 1.20 | |
```
Comment (annajung) [inline content/en/docs/reference/command-line-tools-reference/feature-gates.md:94]: Please provide a high-level description of the feature gate below the table.

**Comment by sftim** [comment]
Hi @derekwaynecarr 

Could you also describe this feature gate - see https://github.com/kubernetes/website/pull/24904#discussion_r529007673?

**Comment by reylejano** [comment]
Hi @derekwaynecarr , just a reminder that the docs merged deadline is on Dec 2. Can you address @sftime 's comment about describing the feature gate. Thank you

**Comment by derekwaynecarr** [comment]
addressed feedback.

**Review by sftim — COMMENTED** [review]
/lgtm

We can't hyperlink to the concept page about the downward API, as there isn't one yet.

**Pull request merged**